python/applications/ranking/data/scripts/create_dataset.py

```diff
@@ -0,0 +1,148 @@
+
+        # Build data
+        example_data = load_example_data(data_dir, logger)
+
+        df_synthetic = fill_data(logger, example_data, max_num_records, feature_config, num_samples)
+        df_synthetic.to_csv(out_file, index=False)
+        logger.info('Synthetic data created! Location: {}'.format(out_file))
+        return df_synthetic
+
+    except Exception as e:
+        logger.error("!!! Error creating synthetic data: !!!\n{}".format(str(e)))
+        traceback.print_exc()
+        return
+
+
+def setup_logging():
+    return logging_utils.setup_logging(reset=True, file_name='test.txt', log_to_file=True)
+
+
+def set_seeds(random_state):
+    # for repeatability
+    np.random.seed(random_state)
+    random.seed(random_state)
+
+
+def load_example_data(data_dir, logger):
+    # Start with loading in CSV format only
+    if file_io.path_exists(data_dir):
+        dfs = file_io.get_files_in_directory(data_dir)
+        return file_io.read_df_list(dfs)
+    else:
+        logger.error("Error! Data directory must exist and be specified")
+
+
+def fill_data(logger, example_data, max_num_records, feature_config, num_samples):
+    """
+    Creates synthetic data using source data as template/optional sampling source
+    Fastest way is to create a list of dicts, then create the DF from the list
+    For the specified number of samples:
+    1. Generate a query key
+    2. Random sample number of results (with replacement)
+    3. Random sample to fill each result (using FeatureConfig)
+    4. TODO Check that no record ID is used twice
+    """
+
+    # Set up new DF and example data distributions for sampling options
+    rows_df = []
+    seed_dict = {}  # the keys are source data columns, the values are lists of source data column values
+    seed_dict_labeled_pos = {}  # option to get characteristics of labeled results separately
+    seed_dict_labeled_neg = {}
+
+    name_num_sequences = [x['serving_info']['name'] for x in feature_config.all_features if x['name']=='num_sequences'][0]
+    name_label_rank = [x['serving_info']['name'] for x in feature_config.all_features if x['name']=='label_rank'][0]
+    name_highval_feature = [x['serving_info']['name'] for x in feature_config.all_features if x['name']=='high_val_feature_1'][0]
+
+    for mycol in list(example_data.columns):
+        # Generate data distributions from sample data
+        seed_dict[mycol] = list(example_data[mycol].values)
+        seed_dict_labeled_pos[mycol] = list(example_data[example_data[feature_config.get_label('name')] == 1].values)
+        seed_dict_labeled_neg[mycol] = list(example_data[example_data[feature_config.get_label('name')] == 0].values)
+
+    seed_dict[name_num_sequences] = list(filter(lambda x: x <= max_num_records, seed_dict[name_num_sequences]))
+
+    for _ in range(num_samples):
+        # Generate a synthetic query ID
+        seed_id = str(random.sample(seed_dict[feature_config.get_query_key('name')], 1)[0])
```

Comment:
@calliebradley While I think this is okay for now, I think it might be more robust if you can just generate a random string or number based on the `dtype` of `query_key`.